### DISCUSSION TITLE
An extension to connect and use bluetooth devices

### DISCUSSION THREAD

**Discussion by BlueToadMakerr** [TurboWarp/extensions#875] [Show and tell]
![image](https://github.com/TurboWarp/extensions/assets/140609260/b805f5b0-5d4b-4c55-b86d-a14e7af3564c)
(You will need the UUID of your Bluetooth device to send and receive data) (Run it unsandboxed) (Will not work on some browsers)

[Bluetooth.js.txt](https://github.com/TurboWarp/extensions/files/12317712/Bluetooth.js.txt)

**Comment by Flappy25** [comment]
Wow! It works perfectly!!

**Comment by ThatBeaverDev** [comment]
you need to make a pull request if you want this added to turbowarp officially

**Reply by BlueToadMakerr** [reply]
How do I do that

**Reply by ThatBeaverDev** [reply]
If you on the web version, go to pull request and open a new request

**Reply by unknown07724** [reply]
@BlueToadMakerr  fork the repo, make a new branch based off of the extensions branch, put a file called [name]/bluetooth.js, then make the pull request and then the mods will deny or accept it

**Comment by BlueDome77** [comment]
Are you going to make a pull request on this? (Aka publish to the gallery)

**Reply by BlueToadMakerr** [reply]
I still don't know how. What do I click
![Screenshot_2023-09-13-15-33-05](https://github.com/TurboWarp/extensions/assets/140609260/d8d31e8a-5bb4-4d43-ae25-cf92df45a2f4)

**Reply by snowboyz0825** [reply]
It's weird. You have to "create a fork" which basically copies the whole repository as your own, put the file in the correct spot with the correct stuff, and make sure it'll pass all checks when merging that repository with this one.

**Reply by BlueToadMakerr** [reply]
how do i merge

**Reply by snowboyz0825** [reply]
that;s what a pull request tells the moderators to do

**Reply by BlueToadMakerr** [reply]
im lost...

**Reply by Thebloxers998** [reply]
Me too (kinda)

**Reply by unknown07724** [reply]
guys, should I ping garbomuffin for your convience

**Comment by Drago-Cuven** [comment]
does it account for multiple bluetooth connected devices?
for example, connecting an xbox one and a ps4 controller to a computer at the same time?

**Reply by Thebloxers998** [reply]
Dunno

**Comment by BlueToadMakerr** [comment]
No. I don't think chrome allows multiple devices to be connected to a site, also it won't connect to the page until you run connect Bluetooth device and find the device to pair

**Comment by Drago-Cuven** [comment]
it does. the gamepad extension could see both controllers which i connected
via bluetooth

**Comment by BlueToadMakerr** [comment]
The gamepad extension does not use chromes Bluetooth API. My extension does.

**Reply by Drago-Cuven** [reply]
still. you said chrome doesn't. it does. that being said using the bluetooth api shouldn't change that fact.

**Comment by Rivercheets** [comment]
how do i get teh bluetooth uuid? also im on windows 11

**Comment by BlueToadMakerr** [comment]
you need to install software that can pull your uuid from your bluetooth
device

**Reply by Rivercheets** [reply]
what is software hname

**Comment by BlueToadMakerr** [comment]
you should be able to look it up for your os

**Reply by Rivercheets** [reply]
how do i look it up on os also what is a uuid? im a bit confused
